package/azure_li_services_spec_template

```diff
@@ -31,6 +31,7 @@ Requires:       python3-PyYAML
 Requires:       python3-setuptools
 Requires:       python3-psutil
 Requires:       python3-humanfriendly
+Requires:       python3-Cerberus
```

Comment:
So I looked up `python3-Cerberus` (I hadn't seen that before). Pretty cool to have that sort of capability built in.

But in my reading, it seemed like you gave it a bunch of test cases and then it returned `true` or `false` if it was valid or not. But I don't see a bunch of test cases here. I would have expected positive/negative test cases for each section.

Am I missing something here, or are the schema validation unit tests a little on the "weak" side?

Reply:
I have only one example for an invalid yaml in the unit tests, this is correct. I did not see a reason to cover more in the scope of this tests. That's because test/data/config.yaml forms a complete and valid configuration. That configuration is read and validated by any of the individual service unit tests and would fail on any change of the schema. Thus there is no reason to have extra tests for specific sections except for the one having no sections at all

If for example the schema is changed on the networking section the test in test/unit/units/network_test.py will fail if that change has an impact on the validation and/or the code. This forces us to update test/data/config.yaml and the code to follow the change and that's what
unit tests are good for.

I did not add tests for the schema itself because I trust cerberus to perform the validation correctly. So unit testing is limited to the project code and not extended to imported code

Do you still think schema validation it not covered good enough ?